### PR TITLE
[E] Restructure

### PR DESCRIPTION
**Pull request by mmatongo**

### CODE REVIEW

**Comment by mmatongo** [comment]
see #17

File: internal/document/docx_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	w.Close()
+	return bytes.NewReader(buf.Bytes())
+}
+
+func createEmptyDocx() io.Reader {
+	return createDocxWithContent(`<?xml version="1.0" encoding="UTF-8"?><document></document>`)
+}
+
+func createSingleParagraphDocx(content string) io.Reader {
+	return createDocxWithContent(`<?xml version="1.0" encoding="UTF-8"?><document><p>` + content + `</p></document>`)
```
Comment (ccoVeille) [inline internal/document/docx_test.go:28]: Is there no risk any character that is invalid in XML to break everything.

Something like

Content = `5 > 3`

File: internal/document/docx_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ProcessDocx() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestProcessDocx_Error_ReadAll(t *testing.T) {
+	_, err := processPptxContent(&errorReader{})
+	assert.NotNil(t, err, "ProcessDocx() did not return an error")
```
Comment (ccoVeille) [inline internal/document/docx_test.go:82]:
```suggestion
	assert.NoErr(t, err, "ProcessDocx() did not return an error")
```

File: internal/document/pdf_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/mmatongo/chew/internal/common"
+)
+
+func getRootPath() string {
+	pwd, _ := os.Getwd()
```
Comment (ccoVeille) [inline internal/document/pdf_test.go:15]: What about this?

```suggestion
func getRootPath(t *testing.T) string {
	t.Helper()
	pwd, err := os.Getwd()
	if err != nil {
		t.Fatalf("getting current folder: %s", err)
	}
```

File: internal/document/pptx_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	w.Close()
+	return bytes.NewReader(buf.Bytes())
+}
+
+func createEmptyPptx() io.Reader {
+	return createPptxWithContent(`<?xml version="1.0" encoding="UTF-8"?><document></document>`)
+}
+
+func createSingleParagraphPptx(content string) io.Reader {
+	return createPptxWithContent(`<?xml version="1.0" encoding="UTF-8"?><document><p>` + content + `</p></document>`)
```
Comment (ccoVeille) [inline internal/document/pptx_test.go:28]: Same question about characters that break XML

File: internal/utils/gcs/gcs_utils.go [hunk 5]
```diff
@@ -37,30 +38,30 @@ func uploadToGCS(ctx context.Context, client *storage.Client, bucket, filename s
...
-func deleteFromGCS(ctx context.Context, client *storage.Client, bucket, objectName string) error {
+func DeleteFromGCS(ctx context.Context, client *storage.Client, bucket, objectName string) error {
 	if err := client.Bucket(bucket).Object(objectName).Delete(ctx); err != nil {
 		return fmt.Errorf("failed to delete object from GCS: %w", err)
 	}
 	return nil
 }
 
-func createStorageClient(ctx context.Context, opts TranscribeOptions) (*storage.Client, error) {
+func CreateStorageClient(ctx context.Context, opts common.TranscribeOptions) (*storage.Client, error) {
```
Comment (ccoVeille) [inline internal/utils/gcs/gcs_utils.go:48]: I would have used


```suggestion
func NewStorageClient(ctx context.Context, opts common.TranscribeOptions) (*storage.Client, error) {
```

It sounds more usual for a constructor

File: internal/utils/gcs/gcs_utils.go [hunk 6]
```diff
@@ -11,10 +11,11 @@ import (
 	speech "cloud.google.com/go/speech/apiv1"
 	"cloud.google.com/go/speech/apiv1/speechpb"
 	"cloud.google.com/go/storage"
+	"github.com/mmatongo/chew/internal/common"
 	"google.golang.org/api/option"
 )
 
-func uploadToGCS(ctx context.Context, client *storage.Client, bucket, filename string) (string, error) {
+func UploadToGCS(ctx context.Context, client *storage.Client, bucket, filename string) (string, error) {
```
Comment (ccoVeille) [inline internal/utils/gcs/gcs_utils.go:18]:
```suggestion
func Upload(ctx context.Context, client *storage.Client, bucket, filename string) (string, error) {
```

You are already in gcs package now

File: internal/utils/gcs/gcs_utils.go [hunk 7]
```diff
@@ -37,30 +38,30 @@ func uploadToGCS(ctx context.Context, client *storage.Client, bucket, filename s
 	return fmt.Sprintf("gs://%s/%s", bucket, objectName), nil
 }
 
-func deleteFromGCS(ctx context.Context, client *storage.Client, bucket, objectName string) error {
+func DeleteFromGCS(ctx context.Context, client *storage.Client, bucket, objectName string) error {
```
Comment (ccoVeille) [inline internal/utils/gcs/gcs_utils.go:41]:
```suggestion
func Delete(ctx context.Context, client *storage.Client, bucket, objectName string) error {
```

Same remark

File: internal/utils/gcs/gcs_utils.go [hunk 8]
```diff
@@ -37,30 +38,30 @@ func uploadToGCS(ctx context.Context, client *storage.Client, bucket, filename s
...
+func CreateStorageClient(ctx context.Context, opts common.TranscribeOptions) (*storage.Client, error) {
 	var clientOpts []option.ClientOption
 	if opts.CredentialsJSON != nil {
 		clientOpts = append(clientOpts, option.WithCredentialsJSON(opts.CredentialsJSON))
 	}
 	return storage.NewClient(ctx, clientOpts...)
 }
 
-func createSpeechClient(ctx context.Context, opts TranscribeOptions) (*speech.Client, error) {
+func CreateSpeechClient(ctx context.Context, opts common.TranscribeOptions) (*speech.Client, error) {
```
Comment (ccoVeille) [inline internal/utils/gcs/gcs_utils.go:56]:
```suggestion
func NewSpeechClient(ctx context.Context, opts common.TranscribeOptions) (*speech.Client, error) {
```

Same

File: internal/utils/gcs/gcs_utils.go [hunk 9]
```diff
@@ -37,30 +38,30 @@ func uploadToGCS(ctx context.Context, client *storage.Client, bucket, filename s
...
 	var clientOpts []option.ClientOption
 	if opts.CredentialsJSON != nil {
 		clientOpts = append(clientOpts, option.WithCredentialsJSON(opts.CredentialsJSON))
 	}
 	return speech.NewClient(ctx, clientOpts...)
 }
 
-func createRecognitionRequest(opts TranscribeOptions, audioInfo *speechpb.RecognitionConfig, gcsURI string) *speechpb.LongRunningRecognizeRequest {
+func CreateRecognitionRequest(opts common.TranscribeOptions, audioInfo *speechpb.RecognitionConfig, gcsURI string) *speechpb.LongRunningRecognizeRequest {
```
Comment (ccoVeille) [inline internal/utils/gcs/gcs_utils.go:64]:
```suggestion
func NewRecognitionRequest(opts common.TranscribeOptions, audioInfo *speechpb.RecognitionConfig, gcsURI string) *speechpb.LongRunningRecognizeRequest {
```

Same

File: internal/utils/utils.go [hunk 10]
```diff
@@ -64,3 +65,69 @@ func ExtractTextFromXML(file *zip.File) ([]string, error) {
...
+/*
+Wondering if this is even necessary but I can see how it can be useful
+as it also removes links, images, and code blocks.
+
+I'm not sure if this is the best way to remove markdown syntax.
+Inspired by https://github.com/mmatongo/site/blob/master/cmd/dnlm/helpers.go#L62-L87
+*/
+
+/* RemoveMarkdownSyntax removes markdown syntax from a string */
+func RemoveMarkdownSyntax(text string) string {
```
Comment (ccoVeille) [inline internal/utils/utils.go:78]: Can't you find a Markdown lib for that?

Something community driven and tested?

File: internal/utils/utils.go [hunk 11]
```diff
@@ -64,3 +65,69 @@ func ExtractTextFromXML(file *zip.File) ([]string, error) {
...
+	patterns := []string{
+		"(```[\\s\\S]*?```)",                      // Code blocks
+		"(`[^`\n]+`)",                             // Inline code
+		"!\\[([^\\]]*?)\\]\\(([^)]+)\\)",          // Images
+		"\\[([^\\]]+)\\]\\(([^)]+)\\)",            // Links
+		"(__|\\*\\*|_|\\*)(.+?)(__|\\*\\*|_|\\*)", // Bold and Italic
```
Comment (ccoVeille) [inline internal/utils/utils.go:84]: Using string literals (backtick notation) could help with the escaped characters here

File: internal/utils/utils.go [hunk 12]
```diff
@@ -64,3 +65,69 @@ func ExtractTextFromXML(file *zip.File) ([]string, error) {
...
+	for _, pattern := range patterns {
+		re := regexp.MustCompile("(?m)" + pattern)
+		switch {
+		case strings.HasPrefix(pattern, "(```"):
+			text = re.ReplaceAllString(text, "$1")
+		case strings.HasPrefix(pattern, "(`"):
+			text = re.ReplaceAllString(text, "$1")
+		case strings.HasPrefix(pattern, "!\\["):
+			text = re.ReplaceAllString(text, "$1 ($2)")
+		case strings.HasPrefix(pattern, "\\["):
```
Comment (ccoVeille) [inline internal/utils/utils.go:102]:
```suggestion
		case strings.HasPrefix(pattern, `\[`):
```

Here (and for most other  `case` statement in the `switch`, the string literal notation would improve the readability

File: internal/document/docx_test.go (hunk 2 above)
Comment (ccoVeille) [inline internal/document/docx_test.go:82]: It's fun that you are almost not using testify in other _test.go files

**Review by ccoVeille — COMMENTED** [review]
A few suggestions

File: internal/document/docx_test.go (hunk 1 above)
Comment (mmatongo) [reply internal/document/docx_test.go:28]: We don't actually do any checks for this to technically no, invalid xml would not break anything, are you suggesting we check to ensure that the xml is correct?

File: internal/document/docx_test.go (hunk 2 above)
Comment (mmatongo) [reply internal/document/docx_test.go:82]: Did you mean `NoError`

File: internal/utils/utils.go (hunk 10 above)
Comment (mmatongo) [reply internal/utils/utils.go:78]: I could but considering that this isn't even used at all I don't see any need to invest actual effort

File: internal/utils/utils.go (hunk 11 above)
Comment (mmatongo) [reply internal/utils/utils.go:84]: Thanks

File: internal/utils/utils.go (hunk 12 above)
Comment (mmatongo) [reply internal/utils/utils.go:102]: I'll make a chore to clean this up

File: internal/document/docx_test.go (hunk 2 above)
Comment (mmatongo) [reply internal/document/docx_test.go:82]: You're right, mostly remnants from the old structure cause I just copied and pasted

File: internal/document/pdf_test.go (hunk 3 above)
Comment (mmatongo) [reply internal/document/pdf_test.go:15]: done, thanks

File: internal/document/docx_test.go (hunk 2 above)
Comment (mmatongo) [reply internal/document/docx_test.go:82]: I've gotten rid of assert and this is now handled differently

File: internal/document/docx_test.go (hunk 1 above)
Comment (ccoVeille) [reply internal/document/docx_test.go:28]: If it cannot fail as the content is supposed to be a valid XML, there is nothing to do I think

File: internal/utils/utils.go (hunk 10 above)
Comment (ccoVeille) [reply internal/utils/utils.go:78]: Ok

**Comment by ccoVeille** [comment]
I see you fixed almost everything in #19, thanks

**Pull request merged**